services/app-api/src/handlers/migrate_rate_documents.ts

```diff
@@ -0,0 +1,152 @@
+                    firstRateRelatedDocument
+                formData.rateInfos[1].supportingDocuments =
+                    secondRateRelatedDocument
+                formData.documents = formData.documents.filter(
+                    (doc) =>
+                        !firstRateRelatedDocument.includes(doc) &&
+                        !secondRateRelatedDocument.includes(doc)
+                )
+            }
+            try {
+                const update = await store.updateHealthPlanRevision(
+                    pkgID,
+                    revision.id,
+                    formData
+                )
+                if (isStoreError(update)) {
+                    console.error(
+                        `StoreError updating revision ${
+                            revision.id
+                        }: ${JSON.stringify(update)}`
+                    )
+                    throw new Error('Error updating revision')
+                }
+            } catch (err) {
+                console.error(`Error updating revision ${revision.id}: ${err}`)
+                throw err
+            }
+        } else {
+            const error = `Error decoding formDataProto for revision ${revision.id} in rate migration: ${decodedFormDataProto}`
+            console.error(error)
+            recordException(error, serviceName, 'migrate_rate_documents')
+        }
+    }
+}
+
+export const getDatabaseConnection = async (): Promise<Store> => {
+    const dbURL = process.env.DATABASE_URL
+    const secretsManagerSecret = process.env.SECRETS_MANAGER_SECRET
+
+    if (!dbURL) {
+        console.error('DATABASE_URL not set')
+        throw new Error('Init Error: DATABASE_URL is required to run app-api')
+    }
+    if (!secretsManagerSecret) {
+        console.error('SECRETS_MANAGER_SECRET not set')
+    }
+
+    const pgResult = await configurePostgres(dbURL, secretsManagerSecret)
+    if (pgResult instanceof Error) {
+        console.error(
+            "Init Error: Postgres couldn't be configured in data exporter"
+        )
+        throw pgResult
+    } else {
+        console.info('Postgres configured in data exporter')
+    }
+    const store = NewPostgresStore(pgResult)
+
+    return store
+}
+
+export const getRevisions = async (
+    store: Store
+): Promise<HealthPlanRevisionTable[]> => {
+    const result: HealthPlanRevisionTable[] | StoreError =
+        await store.findAllRevisions()
+    if (isStoreError(result)) {
+        console.error(
+            `Error getting revisions from db ${JSON.stringify(result)}`
+        )
+        throw new Error('Error getting records; cannot generate report')
+    }
+    return result
+}
+
+export const main: Handler = async (event, context) => {
+    const store = await getDatabaseConnection()
+
+    const revisions = await getRevisions(store)
+
+    // Get the pkgID from the first revision in the list
+    const pkgID = revisions[0].pkgID
+    if (!pkgID) {
```

Comment:
Curious why we only check the first revision and not all of them.

Reply:
The pkgID is stable across revisions, so this is basically a check that we have well-formed data.  https://github.com/Enterprise-CMCS/managed-care-review/blob/d696f81ef6dc9ce7163124d1a35d4c018d48b220/services/app-api/prisma/schema.prisma#L34